shared/src/lib.rs

```diff
@@ -42,9 +42,20 @@ pub const SCALE_FACTOR: usize = 1;
 #[doc(hidden)]
 pub const SCALE_FACTOR: usize = 4;
```

Comment:
Is the reason for keeping `SCALE_FACTOR`, despite the additions of the new features, that 
  it allows quick experimentation with larger buffer sizes if needed?

Reply:
If we're already in a breaking change, we could replace scale_factor with a max_buffers feature.

Reply:
Adding this; turns out only 2 more places use SCALE_FACTOR (and one of them was completely unused).

I'll add a "very big" variety for the unconstrained systems to replace scale_factor.

Reply:
SCALE_FACTOR is now gone in updates; there is the option to scale everything to 1KiB (which is a sensible maximum at least in CoAP, where doing more would mean fragmenting even on a common IPv6 transport); the quadruple_sizes feature was replaced with a large_buffers feature.

Reply:
Looks great!